src/v/transform/transform_manager.cc

```diff
@@ -467,16 +456,12 @@ model::cluster_transform_report manager<ClockType>::compute_report() const {
         const auto& entry = it->second;
         processor* p = entry.processor();
         auto id = p->ntp().tp.partition;
-        auto state = entry.current_state();
-        if (!state) {
-            continue;
-        }
```

Comment:
Seems like this changes the semantics of the report slightly. I don't have all the context...is that a value neutral change?

Reply:
No its the same, previously it was only null-opt if it had been moved, which only happened when deleting the entry

Reply:
Honestly I should remove these side effects from the constructor/destructor and move them directly to where these things are created